apis/v1alpha1/trail.go

```diff
@@ -0,0 +1,153 @@
+	//
+	//    * arn:aws:kms:us-east-2:123456789012:alias/MyAliasName
+	//
+	//    * arn:aws:kms:us-east-2:123456789012:key/12345678-1234-1234-1234-123456789012
+	//
+	//    * 12345678-1234-1234-1234-123456789012
+	KMSKeyID *string `json:"kmsKeyID,omitempty"`
+	// Specifies the name of the trail. The name must meet the following requirements:
+	//
+	//    * Contain only ASCII letters (a-z, A-Z), numbers (0-9), periods (.), underscores
+	//    (_), or dashes (-)
+	//
+	//    * Start with a letter or number, and end with a letter or number
+	//
+	//    * Be between 3 and 128 characters
+	//
+	//    * Have no adjacent periods, underscores or dashes. Names like my-_namespace
+	//    and my--namespace are not valid.
+	//
+	//    * Not be in IP address format (for example, 192.168.5.4)
+	// +kubebuilder:validation:Required
+	Name *string `json:"name"`
+	// Specifies the name of the Amazon S3 bucket designated for publishing log
+	// files. See Amazon S3 Bucket Naming Requirements (https://docs.aws.amazon.com/awscloudtrail/latest/userguide/create_trail_naming_policy.html).
+	// +kubebuilder:validation:Required
+	S3BucketName *string `json:"s3BucketName"`
+	// Specifies the Amazon S3 key prefix that comes after the name of the bucket
+	// you have designated for log file delivery. For more information, see Finding
+	// Your CloudTrail Log Files (https://docs.aws.amazon.com/awscloudtrail/latest/userguide/cloudtrail-find-log-files.html).
+	// The maximum length is 200 characters.
+	S3KeyPrefix *string `json:"s3KeyPrefix,omitempty"`
+	// Specifies the name of the Amazon SNS topic defined for notification of log
+	// file delivery. The maximum length is 256 characters.
+	SnsTopicName *string `json:"snsTopicName,omitempty"`
+
+	Tags []*Tag `json:"tags,omitempty"`
+}
+
+// TrailStatus defines the observed state of Trail
+type TrailStatus struct {
+	// All CRs managed by ACK have a common `Status.ACKResourceMetadata` member
+	// that is used to contain resource sync state, account ownership,
+	// constructed ARN for the resource
+	// +kubebuilder:validation:Optional
+	ACKResourceMetadata *ackv1alpha1.ResourceMetadata `json:"ackResourceMetadata"`
+	// All CRS managed by ACK have a common `Status.Conditions` member that
+	// contains a collection of `ackv1alpha1.Condition` objects that describe
+	// the various terminal states of the CR and its backend AWS service API
+	// resource
+	// +kubebuilder:validation:Optional
+	Conditions []*ackv1alpha1.Condition `json:"conditions"`
+	// Specifies whether log file integrity validation is enabled.
+	// +kubebuilder:validation:Optional
+	LogFileValidationEnabled *bool `json:"logFileValidationEnabled,omitempty"`
+	// Specifies the ARN of the Amazon SNS topic that CloudTrail uses to send notifications
+	// when log files are delivered. The format of a topic ARN is:
+	//
+	// arn:aws:sns:us-east-2:123456789012:MyTopic
+	// +kubebuilder:validation:Optional
+	SnsTopicARN *string `json:"snsTopicARN,omitempty"`
```

Comment:
Does this need to be regenerated? I know we merged the code-generator update for this initialism?

Reply:
YUp nice catch